### ISSUE TITLE
flutter.plugins.contactsservice.contactsservice.ContactsServicePlugin.getCursor

### ISSUE THREAD

**Issue by Veargan** [lukasgit/flutter_contacts#194]
Getting following error:

java.lang.NullPointerException

flutter.plugins.contactsservice.contactsservice.ContactsServicePlugin.getCursor

```
Fatal Exception: java.lang.RuntimeException: An error occurred while executing doInBackground()
       at android.os.AsyncTask$3.done(AsyncTask.java:355)
       at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
       at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
       at java.util.concurrent.FutureTask.run(FutureTask.java:271)
       at java.util.concurrent.ThreadPoolExecutor.processTask(ThreadPoolExecutor.java:1187)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
       at java.lang.Thread.run(Thread.java:784)
Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'android.database.Cursor android.content.ContentResolver.query(android.net.Uri, java.lang.String[], java.lang.String, java.lang.String[], java.lang.String)' on a null object reference
       at flutter.plugins.contactsservice.contactsservice.ContactsServicePlugin.getCursor(ContactsServicePlugin.java:104)
       at flutter.plugins.contactsservice.contactsservice.ContactsServicePlugin.access$500(ContactsServicePlugin.java)
       at flutter.plugins.contactsservice.contactsservice.ContactsServicePlugin$GetContactsTask.doInBackground(ContactsServicePlugin.java:81)
       at flutter.plugins.contactsservice.contactsservice.ContactsServicePlugin$GetContactsTask.doInBackground(ContactsServicePlugin.java)
       at android.os.AsyncTask$2.call(AsyncTask.java:334)
       at java.util.concurrent.FutureTask.run(FutureTask.java:266)
       at java.util.concurrent.ThreadPoolExecutor.processTask(ThreadPoolExecutor.java:1187)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
       at java.lang.Thread.run(Thread.java:784)
```

**Comment by Veargan** [comment]
This issue is affecting 60% of users in production

**Comment by lukasgit** [comment]
@Veargan @morozevich my plate is full at the moment... can either of you submit a PR for this issue?

**Comment by Faiyyazauxy** [comment]
I think we should move this to kotlin since asynctask has been deprecated. I will try to create a PR myself in some days if not then anybody can follow this link for the alternative of asynctask https://techenum.com/async-task-in-android-is-deprecated/

**Comment by Veargan** [comment]
@lukasgit I can't work with open source

@Faiyyazauxy stats of reproducing (300 users)
36% Android 10
32% Android 9
26% Android 8
6% Android 11

**Comment by brianschardt** [comment]
This is urgent for me too.

**Comment by Faiyyazauxy** [comment]
@Veargan i have fixed this in my PR. The said issue happens if i don't give permission and i try to fetch the contacts. You need to add permission check on button click so that you can alert your user accordingly

**Comment by lukasgit** [comment]
Fixed in PR #208
